parser/src/main/java/me/vertretungsplan/parser/LegionBoardParser.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * substitution-schedule-parser - Java library for parsing schools' substitution schedules
+ * Copyright (c) 2016 Johan v. Forstner
+ * Copyright (c) 2016 Nico Alt
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
+ * If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package me.vertretungsplan.parser;
+
+import me.vertretungsplan.exception.CredentialInvalidException;
+import me.vertretungsplan.objects.Substitution;
+import me.vertretungsplan.objects.SubstitutionSchedule;
+import me.vertretungsplan.objects.SubstitutionScheduleData;
+import me.vertretungsplan.objects.SubstitutionScheduleDay;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.nodes.TextNode;
+import org.jsoup.select.Elements;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Parser for LegionBoard, an open source changes management system for schools.
+ *
+ * More information on the <a href="https://legionboard.github.io">official website</a>
+ * and on its <a href="https://gitlab.com/groups/legionboard">project page on GitLab</a>.
+ */
+public class LegionBoardParser extends BaseParser {
+
+	private JSONObject data;
+
+	/**
+	 * URL of API root of given LegionBoard Heart instance
+	 */
+	private String api;
+
+	/**
+	 * Authentication key used in requests
+	 */
+	private String authKey;
+
+	public LegionBoardParser(SubstitutionScheduleData scheduleData, CookieProvider cookieProvider) {
+		super(scheduleData, cookieProvider);
+		data = scheduleData.getData();
+	}
+
+	public SubstitutionSchedule getSubstitutionSchedule() throws IOException, JSONException, CredentialInvalidException {
+		api = data.getString("api");
+
+		/*
+		 * QUESTION: Where do I implement the SHA-256 hash of "[username]//[password]"?
+		 * See the documentation: https://gitlab.com/legionboard/heart/blob/master/doc/README.md
+		 */
+		authKey = getCredential().getHash();
```

Comment:
For authenticating to the API, you should create an own hashing implementation in this class (such as a `private String getHash(UserPasswordCredential credential)` function), `getHash()` is meant for a different use case.

You can retrieve the username and password by casting the credential to a `UserPasswordCredential`:
```java
    UserPasswordCredential credential = (UserPasswordCredential) credential;
    String username = credential.getUsername();
    String password = credential.getPassword();
```
For implementing the SHA-256 hash, you can basically use the same code as in https://github.com/johan12345/substitution-schedule-parser/blob/master/parser/src/main/java/me/vertretungsplan/objects/credential/BaseCredential.java#L49, just inserting `username + "//" + password` as the data.